hub/api/dataset.py

```diff
@@ -357,6 +359,60 @@ def like(
 
         return destination_ds
 
+    @staticmethod
+    def copy(
+        src: str,
+        dest: str,
+        overwrite: bool = False,
+        creds=None,
+        token=None,
+        num_workers: int = 0,
+        public: bool = False,
+    ):
+        src_ds = hub.load(src, read_only=True)
+        src_storage = src_ds.storage
+
+        dest_storage, cache_chain = get_storage_and_cache_chain(
+            dest,
+            creds=creds,
+            token=token,
+            read_only=False,
+            memory_cache_size=DEFAULT_MEMORY_CACHE_SIZE,
+            local_cache_size=DEFAULT_LOCAL_CACHE_SIZE,
+        )
+
+        if dataset_exists(cache_chain):
+            if overwrite:
+                cache_chain.clear()
+            else:
+                raise DatasetHandlerError(
+                    f"A dataset already exists at the given path ({dest}). If you want to copy to a new dataset, either specify another path or use overwrite=True."
+                )
+
+        def copy_func(keys):
```

Comment:
I would recommend you to create a new cache for each worker (src_storage), our LRUCache doesn't support multithreading, which could lead to issues